rskj-core/src/main/java/co/rsk/db/StateRootHandler.java

```diff
@@ -73,18 +59,6 @@ public void register(BlockHeader executedBlock, Trie executionResult) {
             return;
         }
 
-        if (executedBlock.isGenesis()) {
-            Keccak256 genesisStateRoot = convert(executedBlock, executionResult);
-            stateRootTranslator.put(genesisStateRoot, executionResult.getHash());
-        } else {
-            boolean isRskip85Enabled = activationConfig.isActive(ConsensusRule.RSKIP85, executedBlock.getNumber());
-            if (isRskip85Enabled) {
-                Keccak256 orchidStateRoot = convert(executedBlock, executionResult);
-                stateRootTranslator.put(orchidStateRoot, executionResult.getHash());
-            } else {
-                Keccak256 blockStateRoot = new Keccak256(executedBlock.getStateRoot());
-                stateRootTranslator.put(blockStateRoot, executionResult.getHash());
-            }
-        }
+        stateRootDB.put(executedBlock.getStateRoot(), executionResult.getHash().getBytes());
```

Comment:
Can we flush this db to disk? Sounds like a cheap way to save some headaches

Reply:
good point. I've added an intermediate class `StateRootsStoreImpl` and integrated it into the `BlockChainFlusher` service, so now the flushing and closing should be managed by that service.